### PR TITLE
Round12: model-slices=64 — halve attention slices (SOTA=128, 256 regressed)

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

SOTA uses \`--model-slices 128\`. PR #139 tested \`--model-slices 256\` (double) and regressed — adding more attention slices hurt. The opposite direction — \`--model-slices 64\` (half the SOTA value) — has **never been tested**.

Halving the slice count reduces attention capacity per head but may:
- Reduce overfitting by constraining the model's capacity
- Improve training efficiency / gradient concentration within each slice
- Find a sweet spot between too coarse (64→32 would be extreme) and current SOTA (128)

This is the last untested point in the model-slices sweep: 64 → **128 (SOTA)** → 256 (regressed).

## Current Baseline to Beat

| Metric | Value |
|--------|-------|
| val_abupt | **9.484%** |
| test_abupt | 10.580% |
| val_surface_p | 6.305% |
| val_wall_shear | 10.499% |
| val_volume_p | 7.265% |

W&B run: `d03oghpp`, group: `tay-round9-compound-lr1e4-ema999` (PR #115, thorfinn)

## Single Delta

Only one change from SOTA: `--model-slices 64` (SOTA=128).

All other hyperparameters are identical to SOTA PR #115.

## Experiment Instructions

```bash
torchrun --standalone --nproc_per_node=8 train.py \
  --agent nezuko \
  --optimizer lion --lr 1e-4 --weight-decay 5e-4 \
  --lion-beta1 0.9 --lion-beta2 0.99 \
  --no-compile-model --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 --model-slices 64 \
  --ema-decay 0.999 \
  --wandb-group tay-round12-model-slices-64
```

## Results

| Metric | This Run | SOTA (#115) | Delta |
|--------|----------|-------------|-------|
| val_abupt | | 9.484% | |
| test_abupt | | 10.580% | |
| val_surface_p | | 6.305% | |
| val_wall_shear | | 10.499% | |
| val_volume_p | | 7.265% | |

W&B run id:
W&B group: `tay-round12-model-slices-64`

## Suggested Follow-ups

- If model-slices=64 beats SOTA: try model-slices=32 to continue the sweep downward
- If model-slices=64 regresses: slices sweep is closed; 128 is optimal

### CODE REVIEW

**Comment by morganmcg1** [comment]
@nezuko your next experiment is PR #231 — **Round 12: model-slices=64**.

SOTA uses 128 slices; PR #139 tried 256 (regressed). This tests the other direction — halving slices to 64. Clean single-delta run.

Please run the command in the PR description exactly as written and post your W&B run id + full val/test metrics when training completes.

Target to beat: **val_abupt < 9.484%** (test_abupt 10.580%).

**Comment by morganmcg1** [comment]
STUDENT nezuko:

Found two stale DDP launches (16 train.py procs, started ~30min and ~6min before this invocation, both racing on the same 8 GPUs at ~1.87 it/s — half the expected throughput). Killed all 16 processes and relaunched a single clean torchrun at 16:50 UTC. Will report results once training completes.

Logfile: `logs/model_slices_64_v2.log`

**Comment by morganmcg1** [comment]
ADVISOR: Trajectory check at ep4 (val=15.631%, delta=+6.147% vs SOTA).

Curve so far:
- ep1=58.335%
- ep2=28.502%
- ep3=19.337%
- ep4=15.631%

This is well behind a healthy SOTA-trajectory run (e.g. SOTA-replay at ep4 was ~13.5%). Combined with the prior **slices=256 regression** (Round 11), the slice-count axis is now confirmed well-tuned at 128 in both directions. Allowing this to complete for full curve, but treating as a likely close (negative result on the lever).

Continue to ep9; we'll record the closeout when training finishes.

**Comment by morganmcg1** [comment]
ADVISOR: Your PR is in a CONFLICTING state (mergeStateStatus=DIRTY) and needs a rebase onto the latest `tay` branch.

Also: your ep4 trajectory (val=15.631%, +6.147pp vs SOTA) confirms this is a negative result — model-slices=64 regresses significantly. Combined with the prior slices=256 regression (Round 11), the slice-count axis is confirmed well-tuned at 128 in both directions.

**Action items:**
1. Let your run complete through ep9 for full data collection (we'll close with complete metrics)
2. Rebase your branch while the run continues:
   ```bash
   git fetch origin
   git checkout nezuko/model-slices-64
   git rebase origin/tay
   git push --force-with-lease origin nezuko/model-slices-64
   ```
3. Post your complete ep9 val_abupt and test metrics as a final comment when training finishes

We'll close this PR after collecting the full dataset (negative results are still valuable data).

**Comment by morganmcg1** [comment]
## Results: REGRESSION — Closing

**W&B Run:** `rgxygusk` (group: `tay-round12-model-slices-64`)
**State:** running (rt=252min, step=27412)
**Best val_primary/abupt_axis_mean_rel_l2_pct:** 10.3338%
**vs SOTA (9.291%):** +1.043pp — confirmed regression

### Analysis
model_slices=64 is worse than the baseline (model_slices=128 default). This closes out the model_slices axis in both directions:
- model_slices=256 (Round 11): regression
- model_slices=64 (this run): +1.043pp regression

The default model_slices=128 appears to be well-optimized for this architecture and dataset. Closing as a dead end.

**Decision: CLOSE**
